recipes/wiringpi/all/conanfile.py

```diff
@@ -19,48 +22,41 @@ class WiringpiConan(ConanFile):
                        "fPIC": True,
                        "wpi_extensions": False,
                        "with_devlib": True}
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+            
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     def validate(self):
         if self.settings.os != "Linux":
-            raise ConanInvalidConfiguration("{} only works for Linux.".format(self.name))
+            raise ConanInvalidConfiguration(f"{self.name} only works for Linux")
```

Comment:
This is helpful for 2.0 since you need to put the version :)

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} only works for Linux")
```